src/argilla/client/feedback/schemas/response_values.py

```diff
@@ -0,0 +1,112 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from argilla.client.feedback.schemas.enums import QuestionTypes
+from argilla.pydantic_v1 import (
+    BaseModel,
+    StrictInt,
+    StrictStr,
+    confloat,
+    conint,
+    constr,
+    parse_obj_as,
+    root_validator,
+)
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.schemas.questions import QuestionSchema
+
+
+class RankingValueSchema(BaseModel):
+    """Schema for the `RankingQuestion` response value for a `RankingQuestion`. Note that
+    we may have more than one record in the same rank.
+
+    Args:
+        value: The value of the record.
+        rank: The rank of the record.
+    """
+
+    value: StrictStr
+    rank: Optional[conint(ge=1)] = None
+
+
+class SpanValueSchema(BaseModel):
+    """Schema for the `SpanQuestion` response value for a `SpanQuestion`.
+
+    Args:
+        label: The label value of the span.
+        start: The start of the span.
+        end: The end of the span.
+        score: The score of the span.
+    """
+
+    label: constr(min_length=1)
+    start: conint(ge=0)
+    end: conint(ge=0)
+    score: Optional[confloat(ge=0.0, le=1.0)] = None
+
+    @root_validator
+    def check_span(cls, values):
```

Comment:
I saw we already tested for larger than 0 but do we also validate if the span is within the overall text field, i.e., they are not larger than the used text?

Reply:
We're covering basic data validation from SDK, adding more complex validations is not quite simple to implement for this SDK. Since there are these kinds of validations from the server, and the rest of the suggestions/responses didn't include validations, I would keep it simple for this version.